### ISSUE TITLE
Updated sage.engine for TeXShop

### ISSUE THREAD

**Issue by herbs** [dandrake/sagetex#57]
I have an updated engine with a couple of additional items to ignore for the md5 sum. It is also possible to use a single run of `egrep -v` instead of the additional pipe. Line 23 of sage.engine should be

```
new_sum=$(egrep -v '^( _st_.goboom|print .SageT)|(_st_.current_tex_line|SageTeX pause|SageTeX unpause)' "$sagename" | md5)
```

I hope this is helpful.

Good Luck,
Herb Schulz

**Comment by dimpase** [comment]
Thanks for the suggestion about `new_sum`, but what is `sage.engine`? Is it something TeXShop-specific? I know basically nothing about it, more context will be helpful.

**Comment by kcrisman** [comment]
> Thanks for the suggestion about new_sum, but what is sage.engine? Is it
> something TeXShop-specific? I know basically nothing about it, more context
> will be helpful.
>

Yes, TeXShop has various engine files that make sure everything can be run
automagically for various things like XeTeX, SageTeX, etc.  I believe that
it is in fact included in TeXShop, though I don't recall whether the author
of that program just follows this or whether you have to download it
separately - it's been a while since I updated the engine, since it works
so nicely :-)

**Comment by herbs** [comment]
> On Nov 3, 2021, at 4:47 AM, Dima Pasechnik ***@***.***> wrote:
> 
> 
> Thanks for the suggestion about new_sum, but what is sage.engine? Is it something TeXShop-specific? I know basically nothing about it, more context will be helpful.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> 

Howdy,

The sage.engine is specific to TeXShop. It can be found in ~/Library/TeXShop/Engines/Inactive/Sage and needs to be activated by moving it two folders up, to ~/Library/TeXShop/Engines, and it will then appear in the typeset popup menu in the Source Window (not the Typeset Menu!).

There also is a platexmkrc file in the `Sage and Latexmk' folder in the folder above that automates processing for the latexmk based engines, (pdf/xe/lua)latexmk.

Good Luck,

Herb Schulz
***@***.***

**Comment by dimpase** [comment]
> The sage.engine is specific to TeXShop. It can be found in ~/Library/TeXShop/Engines/Inactive/Sage

Hi - well, we don't contribute anything to TeXShop, it's up to them to take up your suggestion, see
https://pages.uoregon.edu/koch/texshop/bugreports.html

**Comment by kcrisman** [comment]
> The sage.engine is specific to TeXShop. It can be found in
> ~/Library/TeXShop/Engines/Inactive/Sage
>
> Hi - well, we don't contribute anything to TeXShop, it's up to them to
> take up your suggestion, see
> https://pages.uoregon.edu/koch/texshop/bugreports.html
>

See also
https://github.com/sagemath/sagetex/blob/bf38e724c01775c99ea064445c2b58c93f23a924/sagetex.dtx#L369
but I agree with Dima that this makes sense, and Richard is very good about
bug reports.
